src/apps/workspaces/service/check_access.py

```diff
@@ -30,6 +30,18 @@ def __init__(self, session, user_id: uuid.UUID, is_super_admin=False):
         self.user_id = user_id
         self.is_super_admin = is_super_admin
 
+    async def _check_applet_roles_with_limited_account_access(
+        self,
+        applet_id: uuid.UUID,
+        roles: list[Role] | None = None,
+        *,
+        exception=None,
+    ):
+        has_access = await AppletAccessCRUD(self.session).has_any_roles_for_applet(applet_id, self.user_id, roles)
+
+        if not has_access:
+            raise exception or AppletAccessDenied()
+
```

Comment:
This function is identical to `_check_applet_roles` below. Are you planning to customize this version in a future PR, or was the duplication unnecessary? If it was intentional, perhaps you could place a TODO here explaining what's planned for it?

Reply:
yes good comment i will remove it, the reason for this was to do another kind of validation but then it appears that we have already have that in another function so removing this and using the same function would be a better option 👍🏻 
good thinking